### PR TITLE
Add tls support for pgstac

### PR DESCRIPTION
**Pull request by johnnylarner**
## Closes

- #375 

## Description
This change moves the default TLS connection from `NoTls` to a configurable struct provided by a small TLS wrapper package from the `tokio_postgres` ecosystem. This makes rudimentary TLS the default encryption which is an approach taken by the widely used `sqlx` library.

I considered using generics, but things got hairy quite quickly. We could open up the `new` method into the public API or add a `From` implementation to make it possible for users to use their own configurations for the TLS struct.

I'd be open to adding a couple of extra tests if you could advise where.

## Checklist

Delete any checklist items that do not apply (e.g. if your change is minor, it may not require documentation updates).

- [x] Unit tests (ran the existing suite)
- [x] Git history is linear
- [x] Commit messages are descriptive
- [x] (optional) Git commit messages follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Code is formatted (`cargo fmt`)
- [x] `cargo test`
- [x] Changes are added to the CHANGELOG

### CODE REVIEW

**Review by gadomski — APPROVED** [review]
@johnnylarner thanks for the contribution! I like it, makes sense. I did some re-organization to move the `tls` module down into **pgstac**, since it could be useful in a non-server context as well. I also got the generics working, sort-of. I had to switch to **ring** for the crypto provider because **object_store** brings it in, which was breaking the default provider setup.

If you won't mind looking through my changes in 4b16e18883f83c6df74f2f58479ae220b8cf0aa0 to make sure they still fit your needs, that would be appreciated.

File: server/src/backend/pgstac.rs [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
 /// A backend for a [pgstac](https://github.com/stac-utils/pgstac) database.
 #[derive(Clone, Debug)]
-pub struct PgstacBackend {
-    pool: Pool<PostgresConnectionManager<MakeRustlsConnect>>,
+pub struct PgstacBackend<Tls>
+where
+    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
+    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
+    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
+    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
```
Comment (johnnylarner) [inline server/src/backend/pgstac.rs:21]: Looks good to me

File: server/src/backend/pgstac.rs [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+{
+    pool: Pool<PostgresConnectionManager<Tls>>,
 }
 
-impl PgstacBackend {
+impl PgstacBackend<MakeRustlsConnect> {
     /// Creates a new PgstacBackend from a string-like configuration.
     ///
+    /// This will use an unverified tls. To provide your own tls, use
+    /// [PgstacBackend::new_from_stringlike_and_tls].
```
Comment (johnnylarner) [inline server/src/backend/pgstac.rs:30]: Logical

**Comment by johnnylarner** [comment]
Thanks @gadomski !

**Pull request merged**